### PR TITLE
AQI-6199: AnalyteResultsImportProfile

### PR DESCRIPTION
**Pull request by KentMiller**

### CODE REVIEW

**Review by dedwards-AQI — APPROVED** [review]

**Review by JonathanRobertson-AQI — APPROVED** [review]
nice job

File: proto/one/operations/sample/analyte_results_import_file.proto [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    // Path to look for files ("/sample-data").
+    string path = 1;
+
+    // File name pattern ("*.csv").
+    string fileName = 2;
+
+    // Field delimiter (",").
+    string delimiter = 3;
+
+    bool ignoreBlankLines = 4;
```
[the remark this answers is not in the document]
Comment (KentMiller) [reply proto/one/operations/sample/analyte_results_import_file.proto:19]: Sure, I added a comment but to be honest, I don't know right now what this setting is used for. If false, does that mean that a blank line in the CSV is an error?

**Review by AQI-DanG — DISMISSED** [review]

**Comment by IsaakMalers-AQI** [comment]
I have a question:  Are there corresponding C# files to the .proto files that are not typically checked in?

**Comment by AQI-DanG** [comment]
> I have a question: Are there corresponding C# files to the .proto files that are not typically checked in?

The C# files are generated as part of the build process and included in the nuget package.  There is a subModule.bat script file that does the generation and copies the files to the correct location.

**Comment by KentMiller** [comment]
> I have a question: Are there corresponding C# files to the .proto files that are not typically checked in?

There are batch files that build the protocol buffer objects in different languages. I think we currently generate C# and Python objects. After these classes are generated, we add them to our nuget package.

There's an onboarding exercise where you'll work through this process.

File: proto/one/operations/spreadsheet/computation_variable_binding.proto [hunk 2]
```diff
@@ -23,4 +24,7 @@ message ComputationVariableBinding {
 
    // used to determine relative cell range for grouping functions
    string grouping = 8;
+
+   // The columnNumber of the reference column if different from columnNum
+   google.protobuf.UInt32Value referenceColumnNum = 9;
```
Comment (AQI-DanG) [inline proto/one/operations/spreadsheet/computation_variable_binding.proto:29]: Not sure why this is showing up as this was my change that was previously merged

**Review by AQI-DanG — APPROVED** [review]
Just making sure I understand the models.
An ImportProfile has a set of ImportFiles and an ImportFile has a set of ImportTelemetries, is that correct?

**Comment by KentMiller** [comment]
> Just making sure I understand the models. An ImportProfile has a set of ImportFiles and an ImportFile has a set of ImportTelemetries, is that correct?

Yes, here's the confluence page that the protobuf was modelled from. https://aquaticinformatics.atlassian.net/wiki/spaces/OP/pages/422183066/Schema+Configuration+Data+Analysis+Results+Import+Profile

File: proto/one/operations/spreadsheet/computation_variable_binding.proto (hunk 2 above)
Comment (KentMiller) [reply proto/one/operations/spreadsheet/computation_variable_binding.proto:29]: Looks like this is how it is in main and also my branch.

**Pull request closed without merging**